src/integrator.jl

```diff
@@ -244,20 +242,18 @@ function step(
         # Update result
         if FullTraj
             res[i] = z
-        else
-            res[1] = z
         end
         if !isfinite(z)
             # Remove undef
             if FullTraj
-                res = res[isassigned.(Ref(res), 1:n_steps)]
+                resize!(res, i)
```

Comment:
It seemed a bit useless to compute the not-`undef`ed entries if we already know that exactly entries 1 to `i` are assigned some value.